raiden/tests/integration/test_version.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+import tempfile
+import re
+import os
+import shutil
+
+from ethereum import _solidity
+
+from raiden.utils import get_contract_path
+from raiden.exceptions import ContractVersionMismatch
+
+
+def replace_contract_version(file_path, new_version):
+    original = open(file_path, 'r')
+    replaced = tempfile.NamedTemporaryFile()
+    for line in original.readlines():
+        if re.match(r'^\s*string constant public contract_version = "\d+.\d+.[\d+\_]";\s*$', line):
+            line = re.sub(r'\d+\.\d+\.[\d+\_]', new_version, line)
+        replaced.write(line)
+    replaced.flush()
+    shutil.copy2(replaced.name, file_path)
+
+
+class TempSolidityDir:
+    def __init__(self, original_directory):
+        self.name = tempfile.mkdtemp()
+        os.rmdir(self.name)  # directory must not exist when using shutil.copytree()
+        shutil.copytree(original_directory, self.name)
+
+    def __del__(self):
+        if self.name is not None:
+            shutil.rmtree(self.name)
+
+
+@pytest.mark.parametrize('number_of_nodes', [1])
+@pytest.mark.parametrize('channels_per_node', [0])
+@pytest.mark.parametrize('blockchain_type', ['geth'])
+def test_deploy_contract(raiden_network, deploy_client):
+    """Test deploying contract with different version than the one we have set in Registry.sol.
+    This test makes sense only for geth backend, tester uses mocked Registry class.
+    """
+    contract_path = get_contract_path('Registry.sol')
+    #  Create temporary directory to put all files required to compile the changed contract to.
+    #  Why? Solidity uses first 40 characters of the file path as a library symbol.
+    #  It would be nice to just do a copy of 'Registry.sol', replace version and include statements
+    #   and then by path substitution argument of solc set the path to something like
+    #   raiden-contracts=/path/to/your/raiden/source/contracts. But then if the path is too long,
+    #   Python solidity compiler will fail because of duplicate library symbol.
+    #
+    #  NB: user is responsible for deleting the directory created with mkdtemp(),
+    #   thus the __del__ method and wrapper
+    temp_dir = TempSolidityDir(os.path.dirname(contract_path))
```

Comment:
Use pytest's temporary directory for tests:

https://docs.pytest.org/en/latest/tmpdir.html#tmpdir-handling